File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        def palindrome(left, right):
            while(left >= 0 and right < len(s) and s[left] == s[right]):
                left -= 1
                right += 1
            
            return right - left - 1
        
        max1 = 0
        start = 0
        for i in range(len(s)):
            even = palindrome(i, i + 1)
            odd = palindrome(i, i)
            curMax = max(odd, even)

            if curMax > max1:
                max1 = curMax
                start = i - (curMax - 1) // 2
            
        return s[start: max1 + start]

File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int res = 0;

        int size = s.size();
        for(int i = 0; i < size; i++)
        {
            // odd length
            int l = i, r = i;
            while(l >= 0 && r < size && s[l--] == s[r++])
                res++;
            
            // even length
            l = i, r = i + 1;
            while(l >= 0 && r < size && s[l--] == s[r++])
                res++;

        }
        return res;
    }


};
